File: c/pe023.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int sum_factors(int);

int main()
{
  const int MAX = 28123;
  int abundant[MAX];

  int ab_count = 0;
  int i;
  for (i = 1; i < MAX; ++i)
  {
    if (i < sum_factors(i))
      abundant[ab_count++] = i;  
  }
   
  int sum = 0;
  for (i = 1; i < MAX; ++i)
  {
    int addi = 0;
    int j;
    for (j = 0; j < ab_count; ++j)
    {
      if (i < abundant[j])
      {
        addi = 1;
        break;
      }
      int diff = i - abundant[j];
      int klast = 0;
      int k;
      for (k = 0; k < ab_count; ++k)
      {
        if (abundant[k] == diff)
        {
          klast = 1;
          break;
        }
      }
      if (klast)
        break;
    }
    if (addi)
      sum += i;
  }
  printf("%d\n", sum);
  return(0);
}

int sum_factors(int number)
{
  int sum = 1;
  int i;
  for (i = 2; i <= sqrt(number); ++i)
  {
    if (!(number % i))
    {
      int divisor = number / i;
      sum += i;
      if (i != divisor)
        sum += divisor;
    }
  }
  return sum;
}

File: javascript/pe023.js
"use strict"

const MAX = 28123;

function sum_factors(number)
{
  let sum = 1;
  for (let i = 2; i <= Math.sqrt(number); ++i)
  {
    if (!(number % i))
    {
      let divisor = number / i;
      sum += i;
      if (i != divisor)
        sum += divisor;
    }
  }
  return sum;
}

let abundant = new Array(MAX);

let ab_count = 0;
for (let i = 1; i < MAX; ++i)
{
  if (i < sum_factors(i))
    abundant[ab_count++] = i;  
}
 
let sum = 0n;
for (let i = 1; i < MAX; ++i)
{
  let addi = 0;
  for (let j = 0; j < ab_count; ++j)
  {
    if (i < abundant[j])
    {
      addi = 1;
      break;
    }
    let diff = i - abundant[j];
    let klast = 0;
    for (let k = 0; k < ab_count; ++k)
    {
      if (abundant[k] == diff)
      {
        klast = 1;
        break;
      }
    }
    if (klast)
      break;
  }
  if (addi)
    sum += BigInt(i);
}
console.log(Number(sum));
